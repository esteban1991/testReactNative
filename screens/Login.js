import {
    View,
    Text,
    SafeAreaView,
    StyleSheet,
    TextInput,
    Pressable,
} from "react-native";

import React, { useState } from "react";
import AsyncStorage from "@react-native-async-storage/async-storage";
import Logo from '../components/Logo'

const Login = ({ navigation }) => {
    const [username, setUsername] = useState("");

	const handleLogin = () => {
		storeUsername(username);   
		 navigation.navigate("Home");
	};

	const storeUsername = async (value) => {
		try {
			await AsyncStorage.setItem("username", value);
		} catch (err) {
			console.error(err);
		}
	};


    return (
        <SafeAreaView style={styles.loginScreen}>           
                
              
            <View style={styles.loginContainer}>
          
            <Logo style={styles.icon} ></Logo>
                <Text
                    style={{
                        fontSize: 24,
                        fontWeight: "bold",
                        marginBottom: 15,
                        textAlign: "center",
                    }}
                >
                    Login
                </Text>
             
                <View style={{ width: "100%" }}>
                    <TextInput
                        id='inputLog'
                        style={styles.textInput}
                        value={username}
                        onChangeText={(value) => setUsername(value)}
                    />
                </View>
                <Pressable onPress={handleLogin} style={styles.loginButton}>
                    <View>
                        <Text id="btn_id" style={{ color: "#fff", textAlign: "center", fontSize: 16 }}>
                            SIGN IN
                        </Text>
                    </View>
                </Pressable>
            </View>
        </SafeAreaView>
    );
};

export default Login;


const styles = StyleSheet.create({
	loginScreen: {
		flex: 1,
      
	},
	loginContainer: {
		flex: 2,
		padding: 10,
		flexDirection: "column",
		justifyContent: "center",      
        
	},
	textInput: {
		borderWidth: 1,
		width: "100%",
		padding: 12,
		marginBottom: 10,
        borderWidth: 1,
        borderRadius: 10,
        padding: 10,
        
	},
	loginButton: {
		width: 150,
		backgroundColor: "#555555",
		padding: 15,
        borderRadius: 10,
        marginTop:40,
        marginLeft:120      
        
	},
    icon:{
        paddingLeft: 8,
        width: 100,
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'flex-start'

    },

});