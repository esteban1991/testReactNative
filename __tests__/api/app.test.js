import React from "react";
import {
	render,
	waitFor,
	screen,
	fireEvent,
} from "@testing-library/react-native";
import Login from "../../screens/Login";
import AsyncStorage from "@react-native-async-storage/async-storage";
import renderer from "react-test-renderer";
jest.useFakeTimers();



describe("<Login />", () => {
  
	it("Fiding Login", async () => {
		render(<Login />);
		const input = screen.getByText("Login");	
    expect(input).toBeTruthy();
	});

  it("Fiding button", async () => {
		render(<Login />);
		const btn = screen.getByText("SIGN IN");	
    expect(btn).toBeTruthy();
	});

});


