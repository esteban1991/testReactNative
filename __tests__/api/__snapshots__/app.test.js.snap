// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "flex": 2,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "padding": 10,
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "height": 60,
                    "marginBottom": 50,
                    "marginLeft": 180,
                    "width": 60,
                  }
                }
              />
              <Text
                style={
                  {
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  }
                }
              >
                Login
              </Text>
              <View
                style={
                  {
                    "width": "100%",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  style={
                    {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginBottom": 10,
                      "padding": 10,
                      "width": "100%",
                    }
                  }
                  value=""
                />
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#555555",
                    "borderRadius": 10,
                    "marginLeft": 120,
                    "marginTop": 40,
                    "padding": 15,
                    "width": 150,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      {
                        "color": "#fff",
                        "fontSize": 16,
                        "textAlign": "center",
                      }
                    }
                  >
                    SIGN IN
                  </Text>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
